draft/XXXX-configuration_handling/README.md

```diff
@@ -0,0 +1,202 @@
+---
+RFC: XXXX
+Author: Roni Choudhury
+Created: 2020-06-29
+Created: 2020-12-09
+Last-Modified: 2020-12-09
+---
+
+# RFC XXXX: Configuration Handling
+
+One of the core features of our vision for Tinker Engine is the ability to
+accept a kind of "meta-configuration" from which concrete configurations are
+generated, used to drive individual protocol runs, with results aggregated
+together in some way that enables post-hoc data analysis. This RFC lays out some
+ideas for some of the core features such a handler will have, along with an
+overall architecture for configuration hanlding in Tinker Engine.
+
+## Proposals
+
+This RFC will discuss a few concrete proposals. One is the overall architcture
+by which the Tinker Engine runtime receives, processes, and submits
+configuration data to a protocol. Another is the method by which a configuration
+file can express variation in its parameters. A third is a fallback mechanism
+allowing programmatic configuration handling when the declarative methods are
+not sufficient for some use case.
+
+### Overall Architecture
+
+Currently, invoking the `tinker` CLI requires specifying a configuration file,
+but currently Tinker does nothing with this file. The new behavior will be for
+Tinker to load the specified file and parse it as YAML (bearing in mind that
+valid JSON is also valid YAML). If the file cannot be opened, or if the contents
+do not parse properly, Tinker will exit with an error condition.
+
+The YAML content describes a "meta-configuration" (see below) that requires
```

Comment:
What do you mean?

Reply:
Thinking about this a bit more, we could use the hash of the git commit as a parameter to ensure that we can track so that we can identify the exact code used to run the experiment to ensure reproducibility from a code perspective.

Reply:
You could certainly keep a git sha as a config parameter if you wanted to. I'm still not sure what exactly you're commenting on in the RFC text though.

Reply:
To make it part of the standard

Reply:
To make keeping a git sha in the config file as part of the standard? Which git sha, and who would update it? What would it mean for that value to be in the configuration file? If you can provide some concrete detail on what you're suggesting here, that would be helpful.

Reply:
Let move this to another RFC after changing the name of this